### PR TITLE
Integrate livepush v3 and live directives

### PR DESCRIPTION
**Pull request by srlowe**
Integrate livepush v3 and live directives

Change-type: minor
Resolves: #1784
Signed-off-by: Scott Lowe <scott@balena.io>

### CODE REVIEW

File: lib/utils/device/live.ts [hunk 1]
```diff
@@ -105,12 +106,14 @@ export class LivepushManager {
...
-				const dockerfile = buildTask.dockerfile;
-				if (dockerfile == null) {
+				if (buildTask.dockerfile == null) {
 					throw new Error(
 						`Could not detect dockerfile for service: ${serviceName}`,
 					);
 				}
+				const dockerfile = new Dockerfile(
+					new Dockerfile(buildTask.dockerfile).generateLiveDockerfile(),
+				);
```
Comment (srlowe) [inline lib/utils/device/live.ts:116]: Not sure if there is a more elegant way of doing this?

File: lib/utils/device/live.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ import {
 } from './deploy';
 import { BuildError } from './errors';
 import { getServiceColourFn } from './logs';
+import Dockerfile from 'livepush/build/dockerfile';
```
Comment (CameronDiver) [inline lib/utils/device/live.ts:21]:
```suggestion
import { Dockerfile } from 'livepush';
```

File: lib/utils/device/live.ts (hunk 1 above)
Comment (CameronDiver) [reply lib/utils/device/live.ts:116]:
```suggestion
const dockerfile = new Dockerfile(buildTask.dockerfile)
```

The generation happens automatically by default, and the internal representation is replaced with the live one. 
`generateLiveDockerfile` is a bit of a misnomer tbh and you only really need it when you want the string representation of the dockerfile

File: lib/utils/device/live.ts [hunk 3]
```diff
@@ -380,13 +383,17 @@ export class LivepushManager {
 			buildLogs[serviceName] = buildLog;
 			const stageImages = LivepushManager.getMultistageImageIDs(buildLogs);
 
-			instance.livepush = await Livepush.init(
-				buildTask.dockerfile!,
-				buildTask.context!,
-				container.containerId,
-				stageImages[serviceName],
-				this.docker,
+			const dockerfile = new Dockerfile(
```
Comment (CameronDiver) [inline lib/utils/device/live.ts:386]:
```suggestion
			const dockerfile = new Dockerfile(buildTask.dockerfile!)
```

**Pull request merged**